### PR TITLE
Devel/error handling

### PR DESCRIPTION
**Pull request by metelik**
A fix for an unhandled error functiona clause

### CODE REVIEW

File: lib/nerves/utils/http_client.ex [hunk 1]
```diff
@@ -69,14 +69,19 @@ defmodule Nerves.Utils.HTTPClient do
...
+      |> Keyword.merge(Keyword.get(opts, :http_opts, []))
 
     opts = [stream: :self, receiver: self(), sync: false]
     :httpc.request(:get, {String.to_charlist(url), headers}, http_opts, opts, :nerves)
     {:noreply, %{s | url: url, caller: from, opts: opts, progress?: progress?}}
   end
 
+  def handle_info({:http, {_ref, {:error, {:failed_connect, [{:to_address, _address}, _reason]}}}}, s) do
+    GenServer.reply(s.caller, {:error, :failed_connect})
+  end
```
Comment (jjcarstens) [inline lib/nerves/utils/http_client.ex:83]:
```suggestion
  def handle_info({:http, {_ref, {:error, {:failed_connect, _}} = err}}, s) do
    GenServer.reply(s.caller, err)
  end
```
Maybe we should report back more in the error so the user can have a bit more details about the connection failure?

Comment (metelik) [reply lib/nerves/utils/http_client.ex]: re-worked as suggested :)

**Comment by jjcarstens** [comment]
Thanks @metelik!

Can you rebase of main and merge your 2 commits? I'm not sure why its showing 0ea6e22 as a new commit here, but that should already be there and not show in this PR

**Comment by fhunleth** [comment]
Rebased and manually merged.

**Pull request closed without merging**